controller/embedding/util.py

```
@@ -1,5 +1,34 @@
-from submodules.model.business_objects import embedding
+from controller.embedding import connector
+from submodules.model import enums
+from submodules.model.business_objects import agreement, embedding, general
+from util import daemon
 
 
 def has_encoder_running(project_id: str) -> bool:
     return embedding.get_first_running_encoder(project_id) is not None
+
+
+def recreate_embedding(
+    project_id: str, embedding_id: str
+) -> None:
+    old_embedding_item = embedding.get(project_id, embedding_id)
+    agreement_item = agreement.get(project_id, old_embedding_item.agreement_id)
```

what about ones that don't have an agreement item? (e.g. hugginface wont have one)
Also on import this agreement item will not exist if I remember correctly we want to throw an error for them and set the old one to error.

---

Recreate is already solved

Creation of agreements for hugging and python can be excluded/filtered

---

solved